install/nodejs

```diff
@@ -0,0 +1,244 @@
+
+    # Grab previous install's binaries so their symlinks can be properly cleaned up before linking new binaries
+    symlink_paths=(/usr/local/node/bin/*)
+    orig_install=$(ls -al ${install_prefix}/node | awk '{print $11}')
+
+    # For all the binaries in the nodejs_dir, delete corresponding symlinks
+    for spath in "${symlink_paths[@]}"; do
+        local check_file="${install_prefix}/bin/$(basename "${spath}")"
+
+        if [[ -e "${check_file}" && ! -L "${check_file}" ]]; then
+            ewarn "Previous node configuration found! ${check_file}"
+            die "${check_file} does not appear to be a symlink, failing!"
+        fi
+        edebug "Removing previous symlink to ${check_file}"
+        rm -f "${check_file}"
+    done
+
+    # Remove existing dir symlink as it will be replaced by current lts
+    edebug "Removing previous node symlink: ${install_prefix}/node -> ${orig_install}"
+    rm -f "${install_prefix}/node"
+
+    eprogress_kill
+}
+
+
+opt_usage retrieve_sha <<'END'
+Retrieves nodejs release sha for the lts codename
+END
+retrieve_sha()
+{
+    $(opt_parse                                                    \
+        ":lts_codename     c | The codename of the lts to install" \
+        ":sha_output_path  o | The place to store the sha"         \
+    )
+
+    einfo "Retrieving SHASUM256.txt for codename ${lts_codename}"
+    local sha_url="https://nodejs.org/download/release/latest-${lts_codename}/SHASUMS256.txt"
+    edebug "Using url: ${sha_url}"
+
+    # Download SHA256 to determine version
+    eretry efetch --style=einfo "${sha_url}" "${sha_output_path}" | edebug
+}
+
+opt_usage get_latest_node_version <<'END'
+Retrieves the latest semantic version for a given nodejs codename
+END
+get_latest_node_version()
+{
+    $(opt_parse                                                                    \
+        ":lts_codename    c | The lts codename to retrieve the latest version for" \
+        ":sha_output_path o | The path where the sha should be saved"
+    )
+
+    # Retrieve the sha file and apply cleanup
+    retrieve_sha --lts_codename "${lts_codename}" --sha_output_path "${sha_output_path}"
+    trap_add "rm -f ${sha_output_path}"
+
+    try
+    {
+        # Find entry with version, and sanitize
+        grep "node-v" "${sha_output_path}" | head -n 1 | awk -F- '{ print $2 }' | sed 's/v//g'
+    }
+    catch
+    {
+        edebug "$(cat "${sha_output_path}")"
+        die "Failed to retrieve version from ${sha_output_path}"
+    }
+}
+
+opt_usage install_node <<'END'
+Installs the nodejs lts version based on the codename passed in
+END
+install_node()
+{
+    $(opt_parse                                                           \
+        ":install_prefix p | The install prefix"                          \
+        ":lts_codename   c | The codename for the lts version to install"
+    )
+
+    lts_version=$(get_latest_node_version --lts_codename "${nodejs_codename}" --sha_output_path "${default_sha_path}")
```

Comment:
fun ebash trick, if you have option flags like `--lts_codename` you can use the more proper looking `--lts-codename`. opt_parse does that magically for you.

Reply:
Noted, I'll do a consistency pass, I missed a couple things now that I see it in github 🤦 (why is that??)